src/overlays/actors/ovl_En_Jso/z_en_jso.c

```diff
@@ -81,106 +159,1546 @@ ActorInit En_Jso_InitVars = {
+    this->actor.speed = 10.0f;
+    this->actor.velocity.y = 20.0f;
+    this->actionFunc = EnJso_SpinBeforeAttack;
+}
+
+/**
+ * Leaps into the air while spinning forward. Once the Garo touches the ground, it will begin a dash attack.
+ */
+void EnJso_SpinBeforeAttack(EnJso* this, PlayState* play) {
+    this->actor.world.rot.x += 0x1770;
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 0xA, 0xFA0, 0x14);
+    if (!(this->actor.velocity.y > 0.0f) && (this->actor.bgCheckFlags & BGCHECKFLAG_GROUND)) {
+        this->actor.world.rot.x = 0;
+        this->actor.velocity.y = 0.0f;
+        this->actor.speed = 0.0f;
+        this->actor.world.rot.y = this->actor.shape.rot.y = this->actor.yawTowardsPlayer;
+        EnJso_SetupDashAttack(this);
+    }
+}
+
+void EnJso_SetupDashAttack(EnJso* this) {
+    this->action = EN_JSO_ACTION_DASH_ATTACK;
+    this->dashAttackTimer = 40;
+    this->bodyCollider.base.colType = COLTYPE_HIT2;
+    this->bodyCollider.base.acFlags &= ~AC_HARD;
+    this->actor.speed = 15.0f;
+    this->actor.velocity.y = 13.0f;
+    Actor_PlaySfx(&this->actor, NA_SE_EN_ANSATSUSYA_ENTRY);
+    Actor_PlaySfx(&this->actor, NA_SE_EN_ANSATSUSYA_DASH_2);
+    this->actionFunc = EnJso_DashAttack;
+}
+
+/**
+ * Dashes toward the player with its swords out. If the player shields the attack, the Garo will be knocked back and
+ * drop its swords. The Garo will stop dashing and perform a slash if it's close enough to the player, if the difference
+ * between its y-rotation and the yaw towards the player is large enough (usually indicating that it has dashed past the
+ * player), or if 40 frames have passed, whichever comes first.
+ */
+void EnJso_DashAttack(EnJso* this, PlayState* play) {
+    f32 curFrame = this->skelAnime.curFrame;
+    s16 yawDiff;
+    s16 absYawDiff;
+    Vec3f knockbackVelocity;
+
+    if ((this->rightSwordCollider.base.atFlags & AT_BOUNCED) || (this->leftSwordCollider.base.atFlags & AT_BOUNCED)) {
+        this->rightSwordCollider.base.atFlags &= ~(AT_HIT | AT_BOUNCED);
+        this->leftSwordCollider.base.atFlags &= ~(AT_HIT | AT_BOUNCED);
+        if (this->swordState == EN_JSO_SWORD_STATE_BOTH_DRAWN) {
+            Matrix_RotateYS(this->actor.yawTowardsPlayer, MTXMODE_NEW);
+            knockbackVelocity.x = 0.0f;
+            knockbackVelocity.y = 0.0f;
+            knockbackVelocity.z = -10.0f;
+            Matrix_MultVec3f(&knockbackVelocity, &this->knockbackVelocity);
+            this->swordState = EN_JSO_SWORD_STATE_KNOCKED_OUT_OF_HANDS;
+            this->dashAttackTimer = 0;
+            this->disableBlure = true;
+            AudioSfx_SetChannelIO(&this->actor.projectedPos, NA_SE_EN_ANSATSUSYA_DASH_2, 0);
+            EnJso_SetupKnockedBack(this);
+            return;
+        }
+    }
+
+    if ((this->actor.velocity.y < 0.0f) && (this->actor.bgCheckFlags & BGCHECKFLAG_GROUND)) {
+        this->actor.velocity.y = 13.0f;
+    }
+
+    if (!(curFrame < this->animEndFrame)) {
```

Comment:
Whenever it comes to theses `!` conditions with floats, I wonder if that's an artificial result of early returns? i.e.
```c
if (curFrame < this->animEndFrame) {
    return;
}
... // rest of the code
```
Just speculation though

Reply:
Every single one matches like this, yeah. I pushed it up in this commit so you can see this change in isolation: https://github.com/zeldaret/mm/pull/1440/commits/464406e817046c173df60aa9f2aa707078797f53

What do you think? If you don't like it, I can just revert, it's no big deal.